### PR TITLE
fix: Add back description column to saved queries #12431

### LINKED ISSUE #12431: [sql_lab]Add back description column to saved queries
**Is your feature request related to a problem? Please describe.**
The description column was removed from the saved queries. We would like to add it back. 
Styling TBD

**Describe the solution you'd like**
Show the description of the saved query if it exists on the saved queries page.

**Describe alternatives you've considered**
Not showing the description, but it seems helpful.

**Additional context**
Banners_and_Alerts_and__DEV__Superset

### PR DESCRIPTION
**Pull request by imancrsrk**
Fixes https://github.com/apache/superset/issues/12431

**Add back Description column**
Screenshot 2024-05-05 at 6 08 34 PM

**Add tooltip for search box**
Screenshot 2024-05-06 at 4 25 32 PM

### CODE REVIEW

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (nytai) [inline superset-frontend/src/pages/SavedQueryList/index.tsx:460]: The search filter should already be checking the description column, so is this really necessary?

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (nytai) [inline superset-frontend/src/pages/SavedQueryList/index.tsx:301]: What happens if there's a really long description?

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (nytai) [reply superset-frontend/src/pages/SavedQueryList/index.tsx:460]: Instead we can rename the "Name" filter to be "Name and Description"

Comment (nytai) [reply superset-frontend/src/pages/SavedQueryList/index.tsx:460]: it's actually searching all text columns (name, description, sql, label), so it can probably just be called "search"

https://github.com/apache/superset/blob/f5843fe588112c5c303215945a36fda0845c9e99/superset/queries/saved_queries/filters.py#L30

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (imancrsrk) [reply superset-frontend/src/pages/SavedQueryList/index.tsx:301]: It's getting truncated which I believe is ok.
Screenshot 2024-05-05 at 7 19 16 PM

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (imancrsrk) [reply superset-frontend/src/pages/SavedQueryList/index.tsx:460]: @nytai I believe the ask is to show query description in a separate second column.

https://github.com/apache/superset/issues/12431#issuecomment-1310895547

**Comment by imancrsrk** [comment]
@nytai All checks have passed. Is it good to merge this ?

File: superset-frontend/src/pages/SavedQueryList/index.tsx
Comment (rusackas) [reply superset-frontend/src/pages/SavedQueryList/index.tsx]: What's the impetus for this change? All the other list views say "Name" since they only search the Name. Does this one also search descriptions? Obviously it ought to be truthful, but otherwise I'd err on the side of consistency.

Comment (imancrsrk) [reply superset-frontend/src/pages/SavedQueryList/index.tsx]: Yes, it searches all text columns (name, description, sql, label), so it can probably just be called "search"

https://github.com/apache/superset/pull/28349#discussion_r1590479993

Comment (rusackas) [reply superset-frontend/src/pages/SavedQueryList/index.tsx]: If you have the wherewithal, it might be nice to add an info tooltip next to the header to state exactly that, so people know exactly what it's searching. Then (on another PR) it'd be good to carry this through to other list views, if any of them indeed search more than name. 

CC @yousoph @kasiazjc in case this merits tracking.

Comment (imancrsrk) [reply superset-frontend/src/pages/SavedQueryList/index.tsx]: ack

Comment (imancrsrk) [reply superset-frontend/src/pages/SavedQueryList/index.tsx]: @rusackas I've added the tooltip & updated the screenshot in description - 

Screenshot 2024-05-06 at 4 25 32 PM

**Comment by imancrsrk** [comment]
@rusackas Hoping you can take another look at the changes we've so far and comment on whether anything else is needed before this pr can be merged and issue resolved?

**Comment by rusackas** [comment]
The tooltip looks strangely high up, otherwise we're looking good, and thanks for the touchup! I'll try to find time in a bit to pull your branch and see about using the tooltip component's props to align it, but I'm working on something else at the moment.

**Comment by imancrsrk** [comment]
> The tooltip looks strangely high up, otherwise we're looking good, and thanks for the touchup! I'll try to find time in a bit to pull your branch and see about using the tooltip component's props to align it, but I'm working on something else at the moment.

ack. I'll try to align it as well later in the evening.

**Comment by imancrsrk** [comment]
> > The tooltip looks strangely high up, otherwise we're looking good, and thanks for the touchup! I'll try to find time in a bit to pull your branch and see about using the tooltip component's props to align it, but I'm working on something else at the moment.
> 
> ack. I'll try to align it as well later in the evening.

@rusackas Hope it looks better now.
Screenshot 2024-05-07 at 9 20 12 PM

File: superset-frontend/src/components/InfoTooltip/index.tsx
Comment (rusackas) [inline superset-frontend/src/components/InfoTooltip/index.tsx:89]: This part seems risky... I'm not sure how many places in the app this would affect. I suspect a lot. @geido do you remember our little alignment tricks here?

Comment (rusackas) [reply superset-frontend/src/components/InfoTooltip/index.tsx:89]: At a minimum, we could set up the InfoTooltip to take an optional iconSize parameter and take it in here. Then, in the other file, instead of `<InfoTooltip tooltip={toolTipDescription} />` you could use `<InfoTooltip tooltip={toolTipDescription} iconSize="m" />`. Then it would only affect the one instance instead of all InfoTooltips in the app.

I swear we had another trick to vertically align these icons, though, besides making them teensy.

Comment (rusackas) [reply superset-frontend/src/components/InfoTooltip/index.tsx:89]: You can avoid altering the InfoTooltip alltogether, and play with the viewbox attribute... it seems like `<InfoTooltip tooltip={toolTipDescription} viewBox='0 -7 28 28'/>` works OK. It's kind of an awkward-feeling attribute to fiddle with, though.

Comment (imancrsrk) [reply superset-frontend/src/components/InfoTooltip/index.tsx:89]: I've pushed this viewBox change. But I'm curious to learn how it works. I tried fiddling with x, y, width, height values but I wasn't able to understand what was going on.

**Comment by michael-s-molina** [comment]
@villebro worked on making all lists consistent (using Name) and may have some input here.

**Comment by rusackas** [comment]
OK, I think we're good to go otherwise... I'll await feedback before unilaterally merging. I think Truthfulness outweighs Consitency, which outweighs pretty much any other random change ;)

**Comment by imancrsrk** [comment]
> OK, I think we're good to go otherwise... I'll await feedback before unilaterally merging. I think Truthfulness outweighs Consitency, which outweighs pretty much any other random change ;)

@rusackas Just following up to find out if we need anything else in this PR or if we are good to merge this change.

**Review by rusackas — APPROVED** [review]
I'm not sure if there's a better way to fix an icon's size/position like this. This _feels_ wrong, but it works without tweaking the underlying component. Still curious if @geido has any better ideas, or if @villebro has any arguments against doing this. Meanwhile, I think it LGTM and will merge if we don't hear back soon.

**Comment by rusackas** [comment]
Done!

**Pull request merged**
